maven-plugin/src/main/java/org/openmrs/maven/plugins/AddExclusion.java

```diff
@@ -51,22 +54,23 @@ public void executeTask() throws MojoExecutionException, MojoFailureException {
             throw new MojoFailureException("Invalid distro file");
         }
 
-        Artifact distroArtifact = originalDistroProperties.getParentArtifact();
-        if (StringUtils.isNotBlank(distroArtifact.getArtifactId()) && StringUtils.isNotBlank(distroArtifact.getGroupId()) && StringUtils.isNotBlank(distroArtifact.getVersion())) {
-            DistroProperties distroProperties = distroHelper.resolveParentArtifact(distroArtifact, new File(distro).getParentFile(), originalDistroProperties, null);
+        Artifact parentDistro = originalDistroProperties.getParentDistroArtifact();
+        if (parentDistro != null) {
+            DistroProperties parentProperties = distroHelper.downloadDistroProperties(new File(distro).getParentFile(), parentDistro);
+            parentProperties = distroHelper.getDistroPropertiesForFullAncestry(parentProperties, new File(distro).getParentFile());
             if (StringUtils.isBlank(property)) {
-                List<String> currentExclusions = distroProperties.getExclusions();
-                List<String> options = distroProperties.getPropertyNames().stream().filter(prop -> !currentExclusions.contains(prop)).collect(Collectors.toList());
-                property = wizard.promptForMissingValueWithOptions("Enter the property you want to exclude",
-                        null, null, options);
-            } else {
-                if (!distroProperties.getPropertyNames().contains(property)) {
-                    wizard.showWarning("The property is not included in the parent distro");
+                List<String> currentExclusions = originalDistroProperties.getExclusions();
+                List<String> options = parentProperties.getPropertyNames().stream().filter(prop -> !currentExclusions.contains(prop)).collect(Collectors.toList());
```

Comment:
Notice the difference here from above - the _options_ need to come from the parent properties, but the current _exclusions_ need to come from the child properties